--- Makefile	2005-02-17 23:01:54.414778896 -0600
+++ Makefile.ok	2005-02-17 23:01:45.558125312 -0600
@@ -60,8 +60,16 @@
 
 Makefile : Makefile.in config.status
 	./config.status
+	patch < Makefile.patch
 	chmod 444 Makefile
 
+Makefile.patch : FORCE
+	rm -f Makefile.patch
+	mv -f Makefile Makefile.ok
+	./configure
+	- diff -u Makefile Makefile.ok > Makefile.patch
+	mv -f Makefile.ok Makefile
+
 config.status: configure
 	./config.status --recheck
 
@@ -81,4 +89,162 @@
 semiclean :
 	rm -f y.tab.* $(LEX_OUTPUT_ROOT).* aonchiall.y rialacha.y
 
+###############################################################################
+#                    REMAINDER IS FOR MAINTAINER ONLY                         #
+###############################################################################
+
+webhome = $(HOME)/public_html/$(PACKAGE_NAME)
+
+installhtml : bugs.html sampla.html FORCE
+	$(INSTALL_DATA) bugs.html $(webhome)
+	$(INSTALL_DATA) cuidiu.html $(webhome)
+	$(INSTALL_DATA) eagar.html $(webhome)
+	$(INSTALL_DATA) foirm.html $(webhome)
+	$(INSTALL_DATA) form.html $(webhome)
+	$(INSTALL_DATA) index.html $(webhome)
+	$(INSTALL_DATA) iompar.html $(webhome)
+	$(INSTALL_DATA) sios.html $(webhome)
+	$(INSTALL_DATA) stair.html $(webhome)
+	$(INSTALL_DATA) sampla.html $(webhome)
+
+installweb : FORCE
+	$(MAKE) installhtml
+	$(INSTALL_DATA) gramadoir.dtd $(webhome)/../dtds
+	$(INSTALL_DATA) gramadoir.lsm $(webhome)
+	$(INSTALL_DATA) gramadoir.vim $(webhome)
+	$(INSTALL_DATA) gramadoir.el $(webhome)
+	$(INSTALL_DATA) OOo/gramadoir.sxw $(webhome)
+
+installcgi : FORCE
+	cp -f gr.cgi /home/httpd/cgi-bin
+	chmod 755 /home/httpd/cgi-bin/gr.cgi
+
+sampla.html : FORCE
+	(cd ga; $(MAKE) triail.html)
+	sed '/gram-ga\.pl triail/r ga/triail.html' samptemplate.html > sampla.html
+
+bugs.html : FORCE
+	(cd ga; $(MAKE) bugs.html)
+	sed '/scannell at slu dot edu/r ga/bugs.html' bugstemplate.html > bugs.html
+
+maintainer-clean :
+	$(MAKE) distclean
+	rm -f ChangeLog configure sampla.html
+
+ChangeLog : FORCE
+	cvs2cl --FSF
+
+dist : FORCE
+	$(MAKE) ChangeLog
+	$(LN_S) gr ../$(TARBALL)
+	tar cvhf $(TARBALL).tar -C .. $(TARBALL)/ABOUT-NLS
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/aonchiall.meta.pl
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/cabhair.pl
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/ChangeLog
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/comhshuite.meta.pl
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/config.guess
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/config.rpath
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/config.sub
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/configure
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/configure.ac
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/COPYING
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/count.pl
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/gettext.h
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/gin.meta.pl
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/gramadoir.dtd
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/gramadoir.el
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/Gramadoir.pm.in
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/gramadoir.vim
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/gramdev.pl
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/gram.pl
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/install-sh
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/Languages-en_us.pm
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/Languages.pm
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/lexbuild
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/Lingua-XX-Gramadoir.t
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/Makefile.in
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/Makefile.PL.in
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/Makefile-xx.in
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/MANIFEST
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/MANIFEST.SKIP
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/messages.txt
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/po2perl.pl
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/ponc.in.l
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/ponc.in.y
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/README
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/README.perl
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/README-xx
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/rialacha.meta.pl
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/OOo/groo.pl
+	tar rvhf $(TARBALL).tar -C .. $(TARBALL)/OOo/gramadoir.sxw
+	cd po && $(MAKE) dist
+	gzip $(TARBALL).tar
+	chmod 444 $(TARBALL).tar.gz
+	rm -f ../$(TARBALL)
+
+sync : FORCE
+	cp -f ABOUT-NLS engine
+	cp -f aonchiall.meta.pl engine
+	cp -f cabhair.pl engine
+	cp -f ChangeLog engine
+	cp -f comhshuite.meta.pl engine
+	cp -f config.guess engine
+	cp -f config.rpath engine
+	cp -f config.sub engine
+	cp -f configure engine
+	cp -f configure.ac engine
+	cp -f COPYING engine
+	cp -f count.pl engine
+	cp -f gettext.h engine
+	cp -f gin.meta.pl engine
+	cp -f gramadoir.dtd engine
+	cp -f gramadoir.el engine
+	cp -f Gramadoir.pm.in engine
+	cp -f gramadoir.vim engine
+	cp -f gramdev.pl engine
+	cp -f gram.pl engine
+	cp -f install-sh engine
+	cp -f Languages-en_us.pm engine
+	cp -f Languages.pm engine
+	cp -f lexbuild engine
+	cp -f Lingua-XX-Gramadoir.t engine
+	cp -f Makefile.in engine
+	cp -f Makefile.PL.in engine
+	cp -f Makefile-xx.in engine
+	cp -f MANIFEST engine
+	cp -f MANIFEST.SKIP engine
+	cp -f messages.txt engine
+	cp -f po2perl.pl engine
+	cp -f ponc.in.l engine
+	cp -f ponc.in.y engine
+	cp -f README engine
+	cp -f README.perl engine
+	cp -f README-xx engine
+	cp -f rialacha.meta.pl engine
+	cp -f OOo/groo.pl engine/OOo
+	cp -f OOo/gramadoir.sxw engine/OOo
+	cp -f po/af.po engine/po
+	cp -f po/boldquot.sed engine/po
+	cp -f po/cy.po engine/po
+	cp -f po/de.po engine/po
+	cp -f po/en@boldquot.header engine/po
+	cp -f po/en@quot.header engine/po
+	cp -f po/fr.po engine/po
+	cp -f po/ga.po engine/po
+	cp -f po/gramadoir.pot engine/po
+	cp -f po/insert-header.sin engine/po
+	cp -f po/LINGUAS engine/po
+	cp -f po/Makefile.in.in engine/po
+	cp -f po/Makevars engine/po
+	cp -f po/mn.po engine/po
+	cp -f po/nl.po engine/po
+	cp -f po/POTFILES.in engine/po
+	cp -f po/quot.sed engine/po
+	cp -f po/remove-potcdate.sin engine/po
+	cp -f po/ro.po engine/po
+	cp -f po/Rules-quot engine/po
+	cp -f po/sk.po engine/po
+	cp -f po/vi.po engine/po
+	cvs commit -m "sync'd with borel repository" engine
+
 FORCE :
