--- Makefile	2006-07-31 17:18:53.000000000 -0500
+++ Makefile.ok	2006-07-31 17:06:36.000000000 -0500
@@ -74,6 +74,15 @@
 
 Makefile : $(GRAMDIR)/Makefile-xx.in configure
 	./configure
+	patch < Makefile.patch
+	chmod 444 Makefile
+
+Makefile.patch : FORCE
+	rm -f Makefile.patch
+	mv -f Makefile Makefile.ok
+	./configure
+	- diff -u Makefile Makefile.ok > Makefile.patch
+	mv -f Makefile.ok Makefile
 
 test : perl
 	(cd $(PERLD); make test)
@@ -182,6 +191,12 @@
 triail : rialacha-$(TEANGA).in
 	sed -n '/^#\./{s/^#\. //; p;}' rialacha-$(TEANGA).in > triail
 
+triailcheck : FORCE
+	cat triail.xml | iconv -f utf8 -t iso-8859-1 | sed 's/^<E offset.*sentence="/<E s="/' > t1.xml
+	$(PERL) -I $(PERLD)/lib $(PERLD)/scripts/$(SCRIPTNAME) --api --iomlan --aspell --comheadan=ga --aschod=iso-8859-1 triail | sed 's/^<E offset.*sentence="/<E s="/' > t2.xml
+	vimdiff t1.xml t2.xml
+	rm -f t1.xml t2.xml
+
 triail.xml-update : FORCE
 	$(MAKE) triail
 	$(MAKE) perl
@@ -269,6 +284,7 @@
 	$(MAKE) rebuildlex
 	rm -f Makefile $(PERLDOK) lexicon-$(TEANGA).txt
 	rm -fR $(PERLD)
+	rm -fR gramadoir
 
 clean :
 	if [ -e $(PERLD)/Makefile ]; then (cd $(PERLD);  make distclean); fi
@@ -315,9 +331,9 @@
 replcheck : FORCE
 	@$(MAKE) perl
 	@echo "earraidi-$(TEANGA).bs..."
-	@sed 's/^[^ ]* //; s/$$/\./' earraidi-$(TEANGA).bs | $(PERL) -I $(PERLD)/lib $(PERLD)/scripts/$(SCRIPTNAME) --litriu --iomlan
+	@sed 's/^[^ ]* //; s/$$/\./' earraidi-$(TEANGA).bs | $(PERL) -I $(PERLD)/lib $(PERLD)/scripts/$(SCRIPTNAME) --litriu --iomlan --aschod=iso-8859-1
 	@echo "eile-$(TEANGA).bs..."
-	@sed 's/^[^ ]* //; s/$$/\./' eile-$(TEANGA).bs | $(PERL) -I $(PERLD)/lib $(PERLD)/scripts/$(SCRIPTNAME) --litriu --iomlan
+	@sed 's/^[^ ]* //; s/$$/\./' eile-$(TEANGA).bs | $(PERL) -I $(PERLD)/lib $(PERLD)/scripts/$(SCRIPTNAME) --litriu --iomlan --aschod=iso-8859-1
 
 
 counts : FORCE
@@ -331,4 +347,101 @@
 zipped : perl
 	(cd $(PERLD); make zipdist)
 
+benchmark : FORCE
+	$(MAKE) perl
+	benchcorp | $(PERL) -I $(PERLD)/lib -d:DProf $(PERLD)/scripts/$(SCRIPTNAME) --iomlan --dath=none --comheadan=ga > /dev/null
+	bash bmtabla >> ../../eile/benchmarks
+	vi ../../eile/benchmarks
+	rm -f tmon.out
+
+unigram-cand.txt : $(DEVSCRIPTNAME)
+	$(MAKE) perl
+	benchcorp | $(PERL) -I $(PERLD)/lib $(DEVSCRIPTNAME) --minic | iconv -f utf8 -t iso-8859-1 > $@
+
+Changes : ../stair.html
+	vilistextum $(GRAMDIR)/stair.html - | sed -n '/^ChangeLog/,/----/p' | $(EGREP) -v -- '---' > $@
+
+# use installed version for this
+triail.html : triail
+	/usr/bin/gram-ga.pl --iomlan --aschod=iso-8859-1 --html --aspell triail > $@
+
+ISPELLDIR = ${HOME}/gaeilge/ispell
+
+# use "groom" instead of this directly (since it requires ispell personal, etc.)
+lexfromdb : FORCE
+	Gin 12
+	((cat $(ISPELLDIR)/ispell-gaeilge/personal $(ISPELLDIR)/ispell-gaeilge/uimhreacha $(ISPELLDIR)/ispell-gaeilge/apost; cat ${HOME}/.biobla | ispell -dgaeilge -e3 | tr " " "\n" | $(EGREP) -v '\/' | aspell -dga -p${HOME}/.aspell.blank.pws list) | sed 's/.*/& 44/'; cat FOCAIL) | $(PERL) -p posmap.pl | LC_COLLATE=POSIX sort -u -k1,1 -k2,2n > lexicon-ga.txt
+	$(MAKE) rebuildlex
+	rm -f FOCAIL
+
+# for convenience (from groom); but also ensures built in correct order
+gafromdb : FORCE
+	$(MAKE) lexfromdb
+	$(MAKE) eilefromdb
+	$(MAKE) earrfromdb
+
+ATH = ${ISPELLDIR}/ispell-gaeilge/athfhocail
+
+# use "groom" instead of this directly
+eilefromdb : FORCE
+	sed 's/ .*//' lexicon-ga.txt > ./tempfocail
+	sed 's/ .*//' $(ATH) | keepif -n ./tempfocail latin-1 | sort -u | join $(ATH) - | LC_COLLATE=POSIX sort -k1,1 | $(EGREP) . > ./eile-ga.bs
+	rm -f tempfocail
+
+EARR = ${ISPELLDIR}/ispell-gaeilge/gaelu ${ISPELLDIR}/ispell-gaeilge/earraidi
+
+# use "groom" instead of this directly
+earrfromdb : FORCE
+	sed 's/ .*//' lexicon-ga.txt eile-ga.bs > ./tempfocail
+	sort -k1,1 $(EARR) > ./tempsrc
+	sed 's/ .*//' ./tempsrc | keepif -n ./tempfocail latin-1 | sort -u | join ./tempsrc - | LC_COLLATE=POSIX sort -k1,1 | $(EGREP) . > ./earraidi-ga.bs
+	rm -f tempfocail tempsrc
+
+lexicon-$(TEANGA).xml : lexicon-$(TEANGA).txt
+	cat lexicon-$(TEANGA).txt | sed 's/ .*//' | LC_ALL=C sort -u | alltags > $@
+
+BIGDIR = /home/spas
+
+big.xml-update : FORCE
+	$(MAKE) perl
+	rm -f $(BIGDIR)/big.xml $(BIGDIR)/bigerr.xml
+	if [ -e $(BIGDIR)/big.new.xml ]; \
+	then \
+		mv $(BIGDIR)/big.new.xml $(BIGDIR)/big.xml; \
+	else \
+		benchcorp | $(PERL) -I $(PERLD)/lib $(PERLD)/scripts/$(SCRIPTNAME) --xml --iomlan --aschod=iso-8859-1 --comheadan=ga | sed 's/<E/\n<E/g; s/<\/E>/&\n/g' | sed '/^<E/!{s/ </\n</g}' | $(EGREP) '.' > $(BIGDIR)/big.xml; \
+	fi
+	if [ -e $(BIGDIR)/bigerr.new.xml ]; \
+	then \
+		mv $(BIGDIR)/bigerr.new.xml $(BIGDIR)/bigerr.xml; \
+	else \
+		$(EGREP) '^<E' $(BIGDIR)/big.xml | LC_ALL=C sort > $(BIGDIR)/bigerr.xml; \
+	fi
+
+bigtest : FORCE
+	@$(MAKE) perl
+	@benchcorp | $(PERL) -I $(PERLD)/lib $(PERLD)/scripts/$(SCRIPTNAME) --xml --iomlan --aschod=iso-8859-1 --comheadan=ga | sed 's/<E/\n<E/g; s/<\/E>/&\n/g' | sed '/^<E/!{s/ </\n</g}' | $(EGREP) '.' | tee $(BIGDIR)/big.new.xml | $(EGREP) '^<E' | LC_ALL=C sort > $(BIGDIR)/bigerr.new.xml
+	@if diff -u $(BIGDIR)/big.xml $(BIGDIR)/big.new.xml > $(BIGDIR)/diff1.txt; then echo "XML unchanged."; rm -f $(BIGDIR)/big.new.xml $(BIGDIR)/diff1.txt; else echo "Differences in XML."; fi
+	@if diff -u $(BIGDIR)/bigerr.xml $(BIGDIR)/bigerr.new.xml > $(BIGDIR)/diff2.txt; then echo "Errors unchanged."; rm -f $(BIGDIR)/bigerr.new.xml $(BIGDIR)/diff2.txt; else echo "Differences in errors."; fi
+
+bugs.html : bugs.txt
+	cat bugs.txt | sed '/^[^#]/s/\*\([^\*]*\)\*/<b class=gramadoir>\1<\/b>/g; s/^[^#].*/<i>&<\/i><br>/; s/^# *HEADING: *\(.*\)/<hr><h2>\1<\/h2><p>/; s/^# *BUG.*/<\/p><p>/; s/^# \(.*\)/\1<br>/;' > $@
+
+testdtd : FORCE
+	$(MAKE) gram-$(TEANGA).dtd
+	cp -f gram-$(TEANGA).dtd ${HOME}/public_html/dtds
+	/usr/bin/gram-ga.pl --iomlan --aschod=utf-8 --xml --aspell triail > temp.xml
+	xmllint --dtdvalid http://borel.slu.edu/dtds/gram-$(TEANGA).dtd temp.xml
+
+gram-ga.exe : FORCE
+	rm -fR gramadoir
+	rm -f gramadoir.zip
+	mkdir gramadoir
+	cp -Rf $(PERLD)/lib/Lingua gramadoir
+	rm -fR gramadoir/Lingua/GA/Gramadoir/Languages*
+	rm -f gramadoir/Lingua/GA/Gramadoir.pm
+#	(cd gramadoir; pp -o gram-ga.exe -l /lib/libc.so.6 -l /lib/libpthread.so.0 /usr/bin/gram-ga.pl)
+	(cd gramadoir; pp -o gram-ga.exe -x ../$(PERLD)/scripts/gram-ga.pl)
+	zip -r gramadoir.zip gramadoir
+
 FORCE :
